browser/brave_ads/request_ads_enabled_api_browsertest.cc

```diff
@@ -0,0 +1,293 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/path_service.h"
+#include "base/test/scoped_feature_list.h"
+#include "brave/browser/brave_ads/ads_service_factory.h"
+#include "brave/common/brave_paths.h"
+#include "brave/components/brave_ads/browser/ads_service.h"
+#include "brave/components/brave_ads/common/features.h"
+#include "brave/components/brave_rewards/browser/test/common/rewards_browsertest_context_util.h"
+#include "brave/components/brave_rewards/browser/test/common/rewards_browsertest_util.h"
+#include "brave/components/brave_rewards/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/network_session_configurator/common/network_switches.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/notification_service.h"
+#include "content/public/browser/notification_types.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "extensions/common/constants.h"
+#include "net/dns/mock_host_resolver.h"
+
+// npm run test -- brave_browser_tests --filter=RequestAdsEnabledApiTest*
+
+namespace {
+
+constexpr char kAllowedDomain[] = "talk.brave.com";
+constexpr char kNotAllowedDomain[] = "brave.com";
+constexpr char kBraveRequestAdsEnabledExists[] =
+    "window.domAutomationController.send("
```

Comment:
these could be raw string literals for better readability

Reply:
Fixed